lib/iex/lib/iex/helpers.ex

```diff
@@ -393,7 +393,14 @@ defmodule IEx.Helpers do
   end
 
   @doc """
-  Prints information about the given data type.
+  Prints information about the data type of a given term.
+  
+  ## Examples
+
+      i([1,2,3])
+      i("Elixir rocks!")
+      i(:some_atom)
+      i(fn x -> x * 2 end)
```

Comment:
@whatyouhide But the example for a list isn't quite so big. How about adding it and just doing something like ```#=> description``` for the others?

```
iex(1)> i [1,2,3]
Term
  [1, 2, 3]
Data type
  List
Reference modules
  List
```